### PR TITLE
MGMT-20474: New route to provide a detailed list of features

### PR DESCRIPTION
**Pull request by pastequo**
* Returns why a feature is not supported
* Also returns, for operators, the list of dependencies

Unit tests have been generated by Cursor

#### Summary (copy paste from CodeRabbit)
- New Features
  - Added a new API endpoint to retrieve detailed information about supported features and operators, including support levels, incompatibilities, reasons, and dependencies, filtered by OpenShift version, CPU architecture, and platform.
  - The API response now provides structured lists of features and operators with explicit incompatibility reasons.
- Bug Fixes
  - Improved validation and error handling for feature and platform support queries.
- Documentation
  - Updated API documentation to describe the new endpoint and response schema.
- Tests
  - Introduced comprehensive tests for feature and platform support logic, including edge cases and validation scenarios.

## List all the issues related to this PR

- [x] New Feature 
- [ ] Enhancement 
- [ ] Bug fix
- [ ] Tests
- [ ] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [ ] Automation (CI, tools, etc)
- [ ] Cloud
- [ ] Operator Managed Deployments
- [x] None

## How was this code tested?

- [ ] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [ ] Waiting for CI to do a full test run
- [ ] Manual (Elaborate on how it was tested)
- [x] No tests needed

## Checklist

- [ ] Title and description added to both, commit and PR.
- [ ] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [ ] This change does not require a documentation update (docstring, `docs`, README, etc)
- [ ] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- Are the title and description (in both PR and commit) meaningful and clear?
- Is there a bug required (and linked) for this change?
- Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

### CODE REVIEW

File: internal/featuresupport/common.go [hunk 1]
```diff
@@ -117,8 +118,49 @@ func IsFeatureCompatibleWithArchitecture(feature models.FeatureSupportLevelID, o
...
+		if externalPlatformName == nil {
+			break
+		}
+
+		switch *externalPlatformName {
+		case common.ExternalPlatformNameOci:
+			supportLevels = append(supportLevels, GetSupportLevel(models.FeatureSupportLevelIDEXTERNALPLATFORMOCI, filters))
+		default:
+			return false, fmt.Errorf("invalid external platform name: %s", *externalPlatformName)
+		}
```
Comment (adriengentil) [inline internal/featuresupport/common.go]: we do support any external platform name. Just that `oci` is a reserved name, and has more restrictions. I think `supportsLevel` should contain `FeatureSupportLevelIDEXTERNALPLATFORM` or `FeatureSupportLevelIDEXTERNALPLATFORMOCI` but not both.

Comment (pastequo) [reply internal/featuresupport/common.go]: changed

**Review by adriengentil — APPROVED** [review]
